### PR TITLE
Added an enum for file attributes on Windows

### PR DESCRIPTION
**Pull request by bausshf**
Also updated relevant documentation for "remove" which may indirectly be tied to "setAttributes" and the "FileAttribute" enum.

This will help people who get "misleading" error messages on Windows when ex. removing a file, because the documentation now specifies what you can do etc.

### CODE REVIEW

File: std/file.d [hunk 1]
```diff
@@ -1008,8 +1009,11 @@ Delete file `name`.
 Params:
     name = string or range of characters representing the file _name
 
-Throws: $(LREF FileException) on error.
- */
+Throws:
+    $(LREF FileException) on error.
+
+    On Windows it may also give an "Access denied" error, which may require you to call $(LREF setAttributes) beforehand with $(LREF FileAttribute).normal
+*/
```
Comment (CyberShadow) [inline std/file.d]: Please do not mix unrelated changes in the same commit / pull request. Remove this change from this pull request, and add it to a separate one.

File: std/file.d [hunk 2]
```diff
@@ -2304,6 +2308,39 @@ if (isConvertibleToString!R)
     }
 }
 
+version (Windows)
+{
+    /// File attributes that can be used on Windows usng ex. $(LREF setAttributes)
```
Comment (CyberShadow) [inline std/file.d]: "usng"

File: std/file.d [hunk 3]
```diff
@@ -2304,6 +2308,39 @@ if (isConvertibleToString!R)
...
+    enum FileAttribute
+    {
+      /// A file or directory that is an archive file or directory. Applications typically use this attribute to mark files for backup or removal.
+      archive = 0x20,
```
Comment (CyberShadow) [inline std/file.d]: Why hard-code these constants instead of reusing the ones from `core.sys.windows`?

File: std/file.d [hunk 4]
```diff
@@ -2304,6 +2308,39 @@ if (isConvertibleToString!R)
     }
 }
 
+version (Windows)
+{
+    /// File attributes that can be used on Windows usng ex. $(LREF setAttributes)
+    enum FileAttribute
+    {
+      /// A file or directory that is an archive file or directory. Applications typically use this attribute to mark files for backup or removal.
```
Comment (CyberShadow) [inline std/file.d]: Phobos style defines indentation to be 4 spaces.

File: std/file.d [hunk 5]
```diff
@@ -2304,6 +2308,39 @@ if (isConvertibleToString!R)
     }
 }
 
+version (Windows)
```
Comment (CyberShadow) [inline std/file.d]: Why only define the Windows attributes?

Comment (CyberShadow) [inline std/file.d]: Windows-only symbol documentation needs to be in a `version(StdDdoc)`, otherwise it won't be visible on dlang.org.

**Comment by RazvanN7** [comment]
ping @bausshf
